src/mds/Server.cc

```diff
@@ -7108,6 +7108,12 @@ void Server::handle_client_link(MDRequestRef& mdr)
   if (target_pin != dir->inode &&
       target_realm->get_subvolume_ino() !=
       dir->inode->find_snaprealm()->get_subvolume_ino()) {
+    if (target_pin->get_parent_inode()->is_stray()) {
+      targeti_pin->add_waiter(CInode::WAIT_UNLINK,
+                              new C_MDS_RetryRequest(mdcache, mdr));
+      mdlog->flush();
```

Comment:
why flush the journal here?

Reply:
Only when the unlink request's journal log is flushed and submitted to the metatata pool will the final step be executed, in which will it unlink the inode and remove the dentry from the parent directory, please see: https://github.com/ceph/ceph/blob/main/src/mds/Server.cc#L7965-L8016

If we don't flush the mdlog, in worst case it may need to wait at most 5 seconds.

Reply:
@lxbsz thanks for the explanation.